jax/_src/debugger/core.py

```diff
@@ -207,4 +213,12 @@ def _breakpoint_callback(*flat_args):
     with debug_lock:
       debugger(frames, thread_id, **kwargs)
 
-  debugging.debug_callback(_breakpoint_callback, *flat_args, ordered=ordered)
+  if token is None:
+    debugging.debug_callback(_breakpoint_callback, *flat_args, ordered=ordered)
+  else:
+    def _breakpoint_callback_wrapper(x, *flat_args):
+      _breakpoint_callback(*flat_args)
+      return x
+    token, flat_args = jax.lax.stop_gradient((token, flat_args))
+    return jax.pure_callback(_breakpoint_callback_wrapper, token, token, *flat_args,
+                             vectorized=True)
```

Comment:
I don't think this matches semantics with the debug callback version. Any reason you're setting vectorized=True here?

Reply:
Agreed this doesn't match the semnatics. I can revert that change for now.

But longer-term I'd like to either add a `vectorized` flag to `jax.debug.breakpoint`, or unconditionally make the `debug_callback` path be vectorized as well. I find it to be far more useful, especally when working with large batches.